l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -390,34 +353,31 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
 
         if (encodingVersion == 0x01) {
             (assetId, transferData) = abi.decode(_data[1:], (bytes32, bytes));
+            require(
+                assetHandlerAddress[assetId] != address(nativeTokenVault),
+                "ShB: new encoding format not yet supported for NTV"
+            );
         } else {
             (assetId, transferData) = _handleLegacyData(_data, _prevMsgSender);
```

Comment:
Please add a comment on why `encodingVersion == 0x01` can not accidentally collide with legacy encoding